scripts/gitutils.py

```diff
@@ -59,19 +59,20 @@ def git_check_merge(branch1, branch2, git=None):
         git.checkout(branch2)
         is_behind = git.log('{0}..{1}'.format(branch2, branch1),
                             max_count=1).strip()
+        clean_merge = True
         if is_behind:
             try:
                 git.merge('--no-commit', '--no-ff', branch1).strip()
             except sh.ErrorReturnCode_1:
                 # git merge returns 1 when there's a conflict
-                return False
-            else:
-                return True
-            finally:
-                git.merge('--abort')
-                git.checkout(orig_branch)
-        else:
-            return True
+                clean_merge = False
+            git.merge('--abort')
```

Comment:
Looks like this unconditionally aborts the merge, and that's a no-op if there is no merge in progress, although it is an error:
```sh
$ git merge --abort
fatal: There is no merge to abort (MERGE_HEAD missing).
$ echo $?
128
```
Should this be indented one more level or possibly put in a separate try/except that catches `sh.ErrorReturnCode`?